### ISSUE TITLE
putScript: request [/_scripts/candidate-score] contains unrecognized parameter: [lang]

### ISSUE THREAD

**Issue by danhlee329** [elastic/elasticsearch-js#805]
- OS: Mac 10.14.3
- npm: 6.4.1
- node: 10.15.0
- elasticsearch client: 15.4.1
- typescript: 3.3.3333
- @types/elasticsearch: 5.0.32

When calling `putScript`, I'm getting this error:
`request [/_scripts/candidate-score] contains unrecognized parameter: [lang]`. Looks like the `lang` parameter is being set as a querystring parameter. I get the same response when replicating the call in Kibana.  When calling the same request in Kibana without the querystring, getting successful response:

```js
{
  "acknowledged": true
}
```

### Call/Logs/Responses

```js
	// Calling elasticsearch client
	this.client.putScript({
		id,
		lang,
		body: {
			script: {
				lang,
 				source
			}
		}
	})

    /* 
		DEBUG LOG
			starting request {"method":"PUT","path":"/_scripts/candidate-score",
			"body":{"script":{"lang":"painless","source":"REDACTED"}},
			"query":{"lang":"painless"}}
    */
```
![esPutScriptsError](https://user-images.githubusercontent.com/7574767/55596584-29886f80-56fe-11e9-96dc-288346454622.png)

What may be causing this issue?

**Comment by spalger** [comment]
Are you setting the `apiVersion` parameter to match your elasticsearch version?

**Comment by danhlee329** [comment]
No I have not, I have not been able to find any info on the `apiVersion` config param.  How can I set that up?

**Comment by danhlee329** [comment]
I've also updated the stack list above, currently using typescript as well.

**Comment by spalger** [comment]
https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/15.x/configuration.html#config-api-version

**Comment by danhlee329** [comment]
Thank you @spalger for the reference. By default, looks like the default value is '6.6'. Initially I have tried to change this value, but still wasn't able to work.  I've looked at the documentation of the API (6.x) and found `putScript` not changed within this major version.

I then checked on the `@types/elasticsearch` package definition (added to the original post above) for [`putStream`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/30e9238945e38fa0f3b3eb7360f0f33cd8f90b9e/types/elasticsearch/index.d.ts#L71) and found that it requires the `lang` property in the [`PutScriptParams`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/30e9238945e38fa0f3b3eb7360f0f33cd8f90b9e/types/elasticsearch/index.d.ts#L505) type, whereas it is not included as a param in the method `putScript` in the API definitions. Bug here?
